### ISSUE TITLE
Oracle Connection Health Check not implemented

### ISSUE THREAD

**Issue by weberk** [eclipse-vertx/vertx-sql-client#1450]
https://github.com/eclipse-vertx/vertx-sql-client/blob/3417c5ab1e346f852a7b9ae99e9a7d71637dbac2/vertx-oracle-client/src/main/java/io/vertx/oracleclient/impl/OracleJdbcConnection.java#L106
io.vertx.oracleclient.impl.OracleJdbcConnection.isValid() returns allways true.
```java
  @Override
  public boolean isValid() {
    return true;
  }
```

Please implement by delegating to the oracle.jdbc.OracleConnection. So calling connection.isValid() you get the JDBC4 compliant result the Oracle JDBC driver delivers, which is much. more efficient than any alternative i.e. calling some table called `dual`!
```java
  @Override
  public boolean isValid() {
    return connection.isValid(oracle.jdbc.OracleConnection.ConnectionValidation.SOCKET, 0);
  }
```

In order to speedup health check using isValid() it might be good to put the following property to the defaults:
```properties
quarkus.datasource.reactive.additional-properties.connectionProperties=oracle.jdbc.defaultConnectionValidation=SOCKET
```

**Comment by tsegismont** [comment]
We can't call JDBC `connection.isValid()` here because it's blocking call. Besides, the expectation from the caller is only to check if the TCP connection is still open.

We have talked in the past about connection validation mechanism when borrowing from the pool, but I can't find any issue.

Does that ring a bell @vietj ?

**Comment by weberk** [comment]
Calling oracle.jdbc.OracleConnection.isValid() is not blocking. For Oracle at least with `ConnectionValidation=SOCKET` it is a quick check of the socket status with the Linux kernel. No remote (blocking) network activity is triggered. Using [lightweight connection validation](https://docs.oracle.com/en/database/oracle/oracle-database/23/jjdbc/JDBC-getting-started.html#GUID-6AFB71F0-EFA9-43A0-AF96-03E6FB2F3521:~:text=Note%3A-,Lightweight%20connection%20validation,-checks%20only%20the) overcomes a limitation of the Java Runtime not consuming Linux signals raised on TCP-FIN events happening on the network. So the Linux kernel knows that the connection is closed, but the Java code has no clue. So the connection pool would hand out a dead connection even though it could have been replaced and some (weak) business logic error handlers are not executed. It improves resilience if isValid() is called.

**Comment by tsegismont** [comment]
I read the [Lightweight Connection Validation](https://docs.oracle.com/en/database/oracle/oracle-database/23/jjdbc/JDBC-getting-started.html#GUID-6AFB71F0-EFA9-43A0-AF96-03E6FB2F3521) section of the docs.

Indeed, it seems that `connection.isValid(oracle.jdbc.OracleConnection.ConnectionValidation.SOCKET, 0);` will do.

However, it doesn't seem necessary to set ther `oracle.jdbc.defaultConnectionValidation` connection property, given that the value is specified in the method call.

**Comment by tsegismont** [comment]
@weberk I've sent #1451 and asked an Oracle specialist some advice. When we're sure about this, the PR can be merged and the changes backported to the `4.x` branch.

**Comment by weberk** [comment]
Thanks for fetching this Thomas! 
You are right 'it is not necessary to set ther oracle.jdbc.defaultConnectionValidation connection property, given that the value specified in the method call is overruling the default.'

**Comment by tsegismont** [comment]
@weberk as discussed in #1451, we can't use the `SOCKET` setting because it may cause the driver to block. In order to provide an experience (and performance) similar to what we have in other clients, we'll use `NONE` (simple connection lifecycle check).

Combined with a pool max lifetime or idle timeout setting, this should make using the Oracle pool more robust.

**Comment by tsegismont** [comment]
Fixed by f08bfedb

**Comment by weberk** [comment]
Thanks for adding any check! Performance is fine, but resilience is also valuable. So now we have a compromise wirh value from both sides. Thanks.
